src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -262,7 +262,14 @@ namespace winrt::TerminalApp::implementation
 
         _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"SelectedItem" });
 
-        if (filteredCommand != nullptr)
+        // Make sure to not send the preview if we're collapsed. This can
+        // sometimes fire after we've been closed, which can trigger us to
+        // preview the action for the empty text (as we've cleared the search
+        // text as a part of closing).
+        const bool isVisible{ this->Visibility() == Visibility::Visible };
```

Comment:
Why not move it into the if condition? That way it can be short-circuited by the compiler.